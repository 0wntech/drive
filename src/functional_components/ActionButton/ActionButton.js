import React from 'react';
import PropTypes from 'prop-types';
import styles from './ActionButton.module.scss';
import classNames from 'classnames';

const ActionButton = ({
    label,
    className,
    color = 'blue',
    onClick,
    size = 'lg',
    type = 'primary',
    disabled,
    dataId,
    children,
}) => {
    return (
        <div
            data-test-id={dataId}
            className={classNames(
                styles.container,
                className,
                styles[color],
                styles[size],
                styles[type],
                {
                    [styles.disabled]: disabled,
                }
            )}
            onClick={onClick}
        >
            {label ? label : children}
        </div>
    );
};

ActionButton.propTypes = {
    color: PropTypes.oneOf(['red', 'green', 'blue', 'white']),
    primary: PropTypes.oneOf(['primary', 'secondary']),
    size: PropTypes.oneOf(['sm', 'md', 'lg']),
    label: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
    disabled: PropTypes.bool,
    dataId: PropTypes.string,
};

export default ActionButton;
