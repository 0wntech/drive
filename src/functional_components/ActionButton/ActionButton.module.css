.container {
    border-radius: 20px;
    color: #000;
    display: flex;
    align-items: center;
    font-family: 'Helvetica Neue Light', 'Helvetica Neue', Helvetica, Arial,
        'Lucida Grande', sans-serif;
}

.blue {
    color: #fff;
    background: #2791ff;
}

.red {
    color: #fff;
    background: #cd0000;
}

.red:hover {
    background: #bd0000;
}

.white {
    background: #fff;
}

.white:hover {
    color: #333;
}

.blue:hover {
    cursor: pointer;
    background-color: #2771ff;
}

.green {
    background: #36c510;
    color: #fff;
}

.green:hover {
    cursor: pointer;
    background: #36a510;
}

.sm {
    font-size: 1em;
    padding: 0.2em 1.5em;
    font-weight: 500;
}

.md {
    font-size: 0.75em;
    padding: 0.1em 1.5em;
    font-weight: 500;
}

.lg {
    font-size: 1em;
    padding: 0.2em 1.5em;
    font-weight: 400;
}

.disabled {
    opacity: 0.5;
    cursor: initial;
}
