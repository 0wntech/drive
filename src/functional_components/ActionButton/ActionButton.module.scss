@import '../../styles/constants.scss';
.container {
    border: 4px solid;
    border-radius: 20px;
    display: flex;
    justify-content: center;
    align-items: center;
    cursor: pointer;
    font-weight: 600;
    box-sizing: border-box;
    color: #fff;
}

.primary {
    background: $light-blue;
    border-color: $dark-blue;
}

.danger {
    background: $light-red;
    border-color: $dark-red;
}

.confirm {
    background: $light_green;
    border-color: $dark_green;
}

.secondary {
    color: $dark-black;
    background: initial;
    border-color: $dark-grey;
}

.sm {
    font: $small-button-text;
    padding: 0 4px;
}

.md {
    font: $medium-button-text;
    padding: 2px 12px;
}

.lg {
    font: $large-button-text;
    padding: 4px 20px;
}

.disabled {
    opacity: 0.5;
    cursor: initial;
}