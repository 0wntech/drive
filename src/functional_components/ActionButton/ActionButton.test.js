import React from 'react';
import ActionButton from './ActionButton';
import { shallow } from 'enzyme';
import TestRenderer from 'react-test-renderer';

describe('ActionButton', () => {
    const label = 'TestButton';

    it('should render default', () => {
        const mockFunction = jest.fn();

        const tree = TestRenderer.create(
            <ActionButton onClick={mockFunction} label={label} />
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should render blue small button', () => {
        const color = 'blue';
        const size = 'sm';
        const mockFunction = jest.fn();

        const tree = TestRenderer.create(
            <ActionButton
                label={label}
                onClick={mockFunction}
                size={size}
                color={color}
            />
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should render medium sized button', () => {
        const mockFunction = jest.fn();

        const size = 'md';
        const tree = TestRenderer.create(
            <ActionButton label={label} onClick={mockFunction} size={size} />
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should trigger onClick function on click', () => {
        const mockFunction = jest.fn();
        const wrapper = shallow(
            <ActionButton label={label} onClick={mockFunction} />
        );
        wrapper.simulate('click');
        expect(mockFunction).toHaveBeenCalled();
    });
});
