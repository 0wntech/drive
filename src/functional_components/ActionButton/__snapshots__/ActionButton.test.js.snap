// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton should render blue small button 1`] = `
<div
  className="container blue sm"
  onClick={[MockFunction]}
>
  TestButton
</div>
`;

exports[`ActionButton should render default 1`] = `
<div
  className="container blue lg"
  onClick={[MockFunction]}
>
  TestButton
</div>
`;

exports[`ActionButton should render medium sized button 1`] = `
<div
  className="container blue md"
  onClick={[MockFunction]}
>
  TestButton
</div>
`;
